homeassistant/components/vacuum/roomba.py

```diff
@@ -0,0 +1,276 @@
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    certificate = config.get(CONF_CERT)
+    continuous = config.get(CONF_CONTINUOUS)
+
+    # Create handler
+    roomba = RoombaVacuum(
+        hass, name, host, username, password, certificate, continuous)
+    hass.data[PLATFORM][host] = roomba
+
+    async_add_devices([roomba], update_before_add=True)
+
+
+class RoombaVacuum(VacuumDevice):
+    """Representation of a Xiaomi Vacuum cleaner robot."""
+
+    def __init__(self, hass, name, host, username, password, certificate,
+                 continuous):
+        """Initialize the Roomba handler."""
+        from roomba import Roomba
+        self.hass = hass
+        self._name = name
+        self._icon = ICON
+        self._host = host
+        self._username = username
+        self._password = password
+        self._certificate = certificate
+        self._continuous = continuous
+        self._battery_level = None
+        self._status = None
+        self._state_attrs = {}
+        self.vacuum_state = None
+        self._is_on = False
+        self._available = False
+        self._metric = hass.config.units.is_metric
+        self.vacuum = Roomba(
+            address=self._host,
+            blid=self._username,
+            password=self._password,
+            cert_name=self._certificate,
+            continuous=self._continuous
+        )
+        _LOGGER.info("Initializing connection with host %s (username: %s)",
+                     self._host, self._username)
+        self.vacuum.connect()
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_ROOMBA
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self._battery_level
+
+    @property
+    def status(self):
+        """Return the status of the vacuum cleaner."""
+        return self._status
+
+    @property
+    def state(self) -> str:
+        """Return the state."""
+        return STATE_ON if self.is_on else STATE_OFF
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self._is_on
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use for device."""
+        return self._icon
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._state_attrs
+
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the vacuum on."""
+        from functools import partial
+        func = partial(self.vacuum.send_command, 'start')
```

Comment:
don't need `partial` to do this. Use it only if you've got keywords, here is just `yield from self.hass.async_add_job(self.vacuum.send_command, 'start')`